tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -884,6 +884,183 @@ export class WebGPUBackend extends KernelBackend {
     }
   }
 
+  private createTexture1Layout(): WebGPULayout {
+    const bindGroupLayoutEntries: GPUBindGroupLayoutEntry[] = [];
+    // Output buffer binding layout.
+    bindGroupLayoutEntries.push({
+      binding: 0,
+      visibility: GPUShaderStage.COMPUTE,
+      storageTexture: {access: 'write-only', format: 'rgba8unorm'}
+    });
+    // Input buffer binding layout.
+    bindGroupLayoutEntries.push({
+      binding: 1,
+      visibility: GPUShaderStage.COMPUTE,
+      buffer: {type: 'read-only-storage' as const}
+    });
+    // Uniform buffer binding layout.
+    bindGroupLayoutEntries.push({
+      binding: 2,
+      visibility: GPUShaderStage.COMPUTE,
+      buffer: {type: 'uniform' as const}
+    });
+    const fromPixelBindGroupLayout =
+        this.device.createBindGroupLayout({entries: bindGroupLayoutEntries});
+    const fromPixelPipelineLayout = this.device.createPipelineLayout(
+        {bindGroupLayouts: [fromPixelBindGroupLayout]});
+    return {
+      bindGroupLayout: fromPixelBindGroupLayout,
+      pipelineLayout: fromPixelPipelineLayout
+    };
+  }
+
+  private runDrawTexture(ctx: GPUCanvasContext, srcTexture: GPUTexture) {
+    const pipeline = this.device.createRenderPipeline({
+      vertex: {
+        module: this.device.createShaderModule({
+          code: `
+          [[stage(vertex)]]
+          fn main([[builtin(vertex_index)]] VertexIndex : u32) -> [[builtin(position)]] vec4<f32> {
+            var pos = array<vec2<f32>, 6>(
+              vec2<f32>(-1.0,  1.0),
+              vec2<f32>(-1.0, -1.0),
+              vec2<f32>( 1.0,  1.0),
+              vec2<f32>(-1.0, -1.0),
+              vec2<f32>( 1.0,  1.0),
+              vec2<f32>( 1.0, -1.0));
+            return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+          }`,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: this.device.createShaderModule({
+          code: `
+[[group(0), binding(0)]] var image0 : texture_2d<f32>;
+[[stage(fragment)]]
+fn main([[builtin(position)]] coord_in: vec4<f32>) -> [[location(0)]] vec4<f32> {
+  var coord_in_vec2 = vec2<i32>(i32(coord_in.x), i32(coord_in.y));
+  let value = textureLoad(image0, coord_in_vec2, 0);
+  //let result = vec4<f32>(value.b, value.g, value.r, value.a);
+  return value;
+}
+          `,
+        }),
+        entryPoint: 'main',
+        targets: [{format: 'bgra8unorm'}],
+      },
+    });
+
+    const uniformBindGroup = this.device.createBindGroup({
+      layout: pipeline.getBindGroupLayout(0),
+      entries: [
+        {
+          binding: 0,
+          resource: srcTexture.createView(),
+        },
+      ],
+    });
+
+    const renderPassDescriptor: GPURenderPassDescriptor = {
+      colorAttachments: [
+        {
+          view: ctx.getCurrentTexture().createView(),
```

Comment:
is this drawing to a different webgpu context? can similar be done with WebGL?

Reply:
This webgpu context is configured to the current using `GPUDevice`. So I can draw to any webgpu context canvas as long as they are set to the same gpu device. You can see that this webgpu context needs to be configured as below:
```
  const gpuContext = canvas.getContext('webgpu');
  gpuContext.configure({
    device: backend.device,
    ...
  });
```
For webgl, you can't draw to a different webgl context. So we have to adjust the default webgl context's size and draw to it. I think webgl should support adjust the default framebuffer's size. I will check it.